operators/pkg/controller/license/license_controller.go

```diff
@@ -144,32 +148,46 @@ func findLicense(c k8s.Client) (v1alpha1.ClusterLicenseSpec, metav1.ObjectMeta,
 func reconcileSecret(
 	c k8s.Client,
 	cluster v1alpha1.Elasticsearch,
-	ref corev1.SecretKeySelector,
-	ns string,
-) (corev1.SecretKeySelector, error) {
-	secretName := cluster.Name + "-license"
-	secretKey := "sig"
-	selector := corev1.SecretKeySelector{
-		LocalObjectReference: corev1.LocalObjectReference{
-			Name: secretName,
-		},
-		Key: secretKey,
-	}
+	clusterLicense v1alpha1.ClusterLicenseSpec,
+	enterpriseLicense metav1.ObjectMeta,
+) error {
+	secretName := esname.LicenseSecretName(cluster.Name)
+	secretKey := "license"
 
 	// fetch the user created secret from the controllers (global) namespace
 	var globalSecret corev1.Secret
-	err := c.Get(types.NamespacedName{Namespace: ns, Name: ref.Name}, &globalSecret)
+	err := c.Get(types.NamespacedName{Namespace: enterpriseLicense.Namespace, Name: clusterLicense.SignatureRef.Name}, &globalSecret)
```

Comment:
If `err` it not used other than to return, it can be folded into the if statement below.